src/compactor_executor.rs

```diff
@@ -123,7 +125,30 @@ impl TokioCompactionExecutorInner {
             .table_store
             .table_writer(SsTableId::Compacted(Ulid::new()));
         let mut current_size = 0usize;
-        while let Some(kv) = all_iter.next_entry().await? {
+        let now = self.options.clock.now();
```

Comment:
Is the change to make this a compaction filter intended for a follow-up PR?

Reply:
we decided to not introduce compaction filters as part of this work because there's a lot of complexity in the API (see discussion #225 and https://github.com/slatedb/slatedb/pull/224#discussion_r1771891263)

or did you just mean to factor it out into a method or something? I can do that

Reply:
I did mean an actual compaction filter not just a method. I'll review that discussion but yeah lets leave it out for now. Thanks for the context!